### PR TITLE
Add fuzzy text search mode

### PR DESCRIPTION
**Pull request by frankh**
Adds a new search mode "fuzzy" which matches as long as the characters in the query exist somewhere in the command in the same order. This is how e.g. SublimeText cmd+P search works

I'm a little worried about performance with large database files. With a ~1000 item database locally it runs no problem

### CODE REVIEW

File: atuin-client/Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ log = "0.4"
 fern = {version = "0.6.0", features = ["colored"] }
 chrono = { version = "0.4", features = ["serde"] }
 eyre = "0.6"
+tokio-test = "*"
```
Comment (conradludgate) [inline atuin-client/Cargo.toml:20]: I think this belongs in `[dev-depenencies]`

File: atuin-client/src/database.rs [hunk 2]
```diff
@@ -286,6 +286,7 @@ impl Database for Sqlite {
         let query = match search_mode {
             SearchMode::Prefix => query,
             SearchMode::FullText => format!("%{}", query),
+            SearchMode::Fuzzy => query.split("").collect::<Vec<&str>>().join("%"),
```
Comment (conradludgate) [inline atuin-client/src/database.rs:289]: The itertools crate (which is already added to this project) provides a `join` method to iterators, without needing to collect them into a vec first: https://docs.rs/itertools/0.10.0/itertools/trait.Itertools.html#method.join

**Review by conradludgate — COMMENTED** [review]
Nice one! Some small changes but it looks good

File: atuin-client/src/database.rs (hunk 2 above)
Comment (conradludgate) [reply atuin-client/src/database.rs:289]: An alternative to using `.split("")` too, is to just use `.chars()`

Comment (frankh) [reply atuin-client/src/database.rs:289]: thanks for the tip with itertools, that's much nicer

split("") actually isn't equivalent to chars() as it adds an empty string at the beginning and end, which we need

Comment (conradludgate) [reply atuin-client/src/database.rs]: Very interesting but also quite unintuitive 🤔 oh well

**Review by conradludgate — APPROVED** [review]

**Comment by ellie** [comment]
Thanks @frankh! 🎉

**Pull request merged**